infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationStreamTableManager.java

```diff
@@ -124,128 +156,79 @@ private void initStreamNameFetcherPlugin() {
         }
     }
 
-    private boolean verifyTableExists(String tableName) {
-        try {
-            corfuStore.getTable(CORFU_SYSTEM_NAMESPACE, tableName);
-        } catch (NoSuchElementException e) {
-            // Table does not exist
-            return false;
-        } catch (IllegalArgumentException e) {
-            // Ignore
-        }
-        return true;
-    }
-
-    private void openExistingTable(String tableName) {
+    private void openTable(String tableName) throws Exception {
         try {
             if (Objects.equals(tableName, LOG_REPLICATION_STREAMS_NAME_TABLE)) {
-                corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, tableName, TableInfo.class,
+                streamsNameTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, tableName, TableInfo.class,
                     Namespace.class, CommonTypes.Uuid.class, TableOptions.builder().build());
             } else {
-                corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LOG_REPLICATION_PLUGIN_VERSION_TABLE, VersionString.class,
+                pluginVersionTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LOG_REPLICATION_PLUGIN_VERSION_TABLE, VersionString.class,
                         Version.class, CommonTypes.Uuid.class, TableOptions.builder().build());
             }
         } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
-            log.warn("Exception when opening existing table ", e);
+            log.warn("Exception when opening table {}", tableName, e);
+            throw e;
         }
     }
 
-    private boolean tableVersionMatchesPlugin() {
-        VersionString versionString = VersionString.newBuilder().setName("VERSION").build();
-        CorfuStoreEntry<VersionString, Version, CommonTypes.Uuid> record;
-
-        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
-            // If the version table is dropped using the CorfuStoreBrowser(UFO) for testing, it will be empty.
-            // In this case, it should be re-created
-            record = txn.getRecord(LOG_REPLICATION_PLUGIN_VERSION_TABLE, versionString);
-            txn.commit();
+    private VersionResult verifyVersionResult(TxnContext txn) {
+        CorfuStoreEntry<VersionString, Version, CommonTypes.Uuid> record =
+                txn.getRecord(LOG_REPLICATION_PLUGIN_VERSION_TABLE, versionString);
+
+        if (record.getPayload() == null) {
+            // Initializing
+            log.info("Version result: INITIALIZING");
```

Comment:
Messages here could be more valuable:
"LR initializing. Version unset"
"LR upgraded. Version changed from X to Y"